### PR TITLE
POC of using background writer.

### PR DESCRIPTION
**Pull request by cmelchior**
Just exposing for visibility. Will need heavy refactoring.

This PR shows how we can use public frozen Realm while delegating writes to a background writer, that when it completes, will update the frozen Realm.

### CODE REVIEW

**Review by rorbech — COMMENTED** [review]
Just some immediate comments before digging into implementing the task thread for Darwin.

File: packages/cinterop/src/darwinCommon/kotlin/io/realm/interop/RealmInterop.kt [hunk 1]
```diff
@@ -150,6 +115,19 @@ fun String.toRString(memScope: MemScope) = cValue<realm_string_t> {
 
 actual object RealmInterop {
 
+    actual fun realm_get_version_id(realm: NativePointer): Pair<ULong, ULong>? {
+        memScoped {
+            val info = alloc<realm_version_id_t>()
+            val found = alloc<BooleanVar>()
+            realm_get_version_id(realm.cptr(), found.ptr, info.ptr)
```
[the remark this answers is not in the document]
Comment (cmelchior) [reply packages/cinterop/src/darwinCommon/kotlin/io/realm/interop/RealmInterop.kt:122]: No, if the version is missing it probably indicate a misuse of the API, but it isn't something that would result in an error from Core, just that things haven't been called in the correct order. 

I was also uncertain exactly how to surface this, but decided to move that decision to the `library`, but e.g. throwing `IllegalStateException` here would probably be fine fine.

File: packages/library/src/iosArm64Main/kotlin/io/realm/internal/worker/getTaskThread.kt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package io.realm.internal.worker
```
[the remark this answers is not in the document]
Comment (cmelchior) [reply packages/library/src/iosArm64Main/kotlin/io/realm/internal/worker/getTaskThread.kt:1]: No, just an import gone wrong

**Comment by cmelchior** [comment]
Replaced by https://github.com/realm/realm-kotlin/pull/291

**Pull request closed without merging**